src/components/common/tab-panel.tsx

```diff
@@ -1,39 +1,47 @@
 "use client";
 
 import Button from "@codegouvfr/react-dsfr/Button";
-import { cloneElement, useState } from "react";
+import { cloneElement, useState, ReactElement } from "react";
+
+// Define more specific types for children where possible for better type safety.
+// For example, if Tab components are the only valid children for Tabs.
+
+interface TabProps {
+  selected?: boolean;
+  disabled?: boolean;
+  handleClick?: () => void;
+  children: string | ReactElement | ReactElement[];
+}
 
 export const Tab = ({
   children,
   selected,
   disabled = false,
   handleClick = () => {},
-}: {
-  selected?: boolean;
-  disabled?: boolean;
-  handleClick?: () => void;
-  children: string | JSX.Element | JSX.Element[];
-}) => (
+}: TabProps) => (
   <div className={`tab${selected ? " selected" : ""}`}>
     <Button
       disabled={disabled}
       onClick={handleClick}
       priority="tertiary no outline"
+      size="sm"
     >
       {children}
     </Button>
   </div>
 );
 
+interface TabsProps {
+  selectedIndex?: number;
+  children: ReactElement<TabProps>[]; // Assuming children are Tab components
+  setSelectedIndex?: (index: number) => void;
+}
+
 export const Tabs = ({
   children,
   selectedIndex = 0,
-  setSelectedIndex = () => {},
-}: {
-  selectedIndex?: number;
-  children: JSX.Element[];
-  setSelectedIndex?: (index: number) => void;
-}) => (
+  setSelectedIndex = ()_ => {},
```

Comment:
Peut-être que ça ça va péter
```suggestion
  setSelectedIndex = () => {},
```